src/assemble.ts

```diff
@@ -77,6 +78,19 @@ export const assemble: AssembleFn = <T>(
           if (child[childIdentifier.split('.')[1]] !== null) {
             parent[parentChildRelation[1]].push(child);
           }
+        } else if (parentChildRelation[0] === 'hasOne') {
+          if (child[childIdentifier.split('.')[1]] !== null) {
```

Comment:
maybe we should put `childIdentifier.split('.')[1]` in a variable as it is being calculated 4 times

Reply:
fixed